test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt

```diff
@@ -1413,21 +1413,69 @@ class QueryTests {
     // --------------
 
     @Test
-    @Ignore
-    fun first_find() {
-        // TODO
+    fun first_find_emptyTable() {
+        realm.query<QuerySample>()
+            .first()
+            .find { first -> assertNull(first) }
     }
 
     @Test
-    @Ignore
-    fun first_find_emptyTable() {
-        // TODO
+    fun first_find() {
+        val value1 = 1
+        val value2 = 2
+
+        realm.writeBlocking {
+            // Queries inside a write transaction produce live results which means they can be
+            // reused within the closure
+            val firstQuery = query<QuerySample>("intField > $0", value1)
+                .first()
+
+            assertNull(firstQuery.find())
+            copyToRealm(QuerySample(intField = value1))
+            copyToRealm(QuerySample(intField = value2))
+            val first = firstQuery.find()
+            assertNotNull(first)
+            assertEquals(value2, first.intField)
+        }
+
+        realm.query<QuerySample>("intField > $0", value1)
+            .first()
+            .find { first ->
+                assertNotNull(first)
+                assertEquals(value2, first.intField)
+            }
     }
 
     @Test
-    @Ignore
     fun first_asFlow() {
```

Comment:
What about the empty table case?

Reply:
https://github.com/realm/realm-kotlin/blob/ce24690ff895881bed9a9827c12000f1b6ae7659/test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt#L1416

Reply:
I missed the `asFlow` part. I have added it now.